### PR TITLE
Rewrite CONFIGURE_SUBMITTY.sh into python

### LINKED ISSUE #1012: Change "tagrading_logs" to be "site_logs"
We log errors and access and other stuff not just from tagrading now but also the site the students see, etc.

We want probably want to rename this to just be "site_logs" or something that makes more sense than "tagrading_logs".

### LINKED ISSUE #1145: Add question to CONFIGURE_SUBMITTY.sh about authentication method
Add question that reads as follows:

```
What authentication method do you want:
1. PamAuthentication
2. DatabaseAuthentication
?
```

allow them to select number, than write that info to INSTALL_SUBMITTY.sh to be written into site/config/master.ini

### PR DESCRIPTION
**Pull request by MasterOdin**
Continues on #918 by porting over CONFIGURE_SUBMITTY.sh. It's currently left in the code base until I've had more people test out the script to make sure there was no bugs introduced.

I also took this occasion to rewrite the log directory structure to close #1012 and enable specifying the authentication method to close #1145.

This additionally gives defaults when asking for input and will reuse the previously entered answers (if available).

### CODE REVIEW

File: .setup/CONFIGURE_SUBMITTY.sh [hunk 1]
```diff
@@ -51,8 +48,8 @@ SUBMITTY_INSTALL_DIR=/usr/local/submitty
 SUBMITTY_DATA_DIR=/var/local/submitty
 
 # Log locations
-TAGRADING_LOG_PATH=$SUBMITTY_DATA_DIR/tagrading_logs/
-AUTOGRADING_LOG_PATH=$SUBMITTY_DATA_DIR/autograding_logs/
+TAGRADING_LOG_PATH=$SUBMITTY_DATA_DIR/logs/site/
```
Comment (bmcutler) [inline .setup/CONFIGURE_SUBMITTY.sh]: Do we need the "site" folder?
I suggest:
/var/local/submitty/logs/autograding
/var/local/submitty/logs/access
/var/local/submitty/logs/error

**Comment by bmcutler** [comment]
(got lost when I removed CONFIGURE_SUBMITTY.sh)

Do we need the "site" folder?
I suggest:
/var/local/submitty/logs/autograding
/var/local/submitty/logs/access
/var/local/submitty/logs/error

**Comment by MasterOdin** [comment]
I'd say yes as just "error" is not clear what it might contain. Should it also contain errors from the autograder? How about errors from rainbow grades? Etc.

**Comment by bmcutler** [comment]
autograding & access are logs of normal activity at the system level.  We might introduce other normal activity logs later.  Autograding logs do contain some error messages within the log -- and I argue those specific errors should stay where they are.

The error log should be empty, if things are debugged and not malfunctioning.  When things start malfunctioning, yes, I think all sorts of detailed errors could be written to that directory -- as a combined log or as separate files, whatever makes sense.

**Comment by bmcutler** [comment]
The autograding logs folder contains info allowing the sysadmin to monitor machine load.  Instructors generally don't use this to debug.  Rather they use the logs per assignment in their course directory.

The rainbow grades errors contain confidential info that should not be at the system level, but rather per course, so the instructor can get in there are tinker as needed.

**Comment by MasterOdin** [comment]
@bmcutler So I've made the following changes to this:
1) flatten `sites/*` to be `site_errors/` and `access/`
2) have CONFIGURE_SUBMITTY.py write out a JSON file with all answers
3) Have CONFIGURE_SUBMITTY.py supply the user with defaults with those defaults being the previously set configuration if file was loaded (so for rerunning CONFIGURE on the live server will just be hitting enter a bunch of times).

**Pull request merged**